### PR TITLE
Add generic function to read file as Woodwork DataFrame

### LINKED ISSUE #851: Add read_file util function to read file given location and optional conten_type
- As a user I wish I could use Woodwork to read a filepath into a dataframe. I wish that Woodwork would determine the type of file (based on the filepath extension), and read the file accordingly. 
- If the content type is not specified, we can use the file extension.
  - If there is no content type and no file extension, then we can use the mimetypes functionality in Python to determine the content type
    - https://docs.python.org/3/library/mimetypes.html
- This function will replace the `read_csv` function in `woodwork/utils.py`

#### Code Example

```python
from woodwork import read_file

df = read_file('data.csv')

# user can also pass content_type if the extension does not exist
df = read_file('data', content_type="text/csv")
```
#### Function
```python
def read_file(filepath=None,
              content_type=None,
              name=None,
              index=None,
              time_index=None,
              semantic_tags=None,
              logical_types=None,
              use_standard_tags=True,
              validate=True,
              **kwargs):
```

### PR DESCRIPTION
**Pull request by gsheni**
- Closes #851

### CODE REVIEW

File: woodwork/tests/test_utils.py [hunk 1]
```diff
@@ -164,11 +164,47 @@ def test_list_semantic_tags():
...
+        ww.read_file(filepath=filepath)
+
+
+def test_read_file_errors_unhandled_content_type(sample_df_pandas, tmpdir):
+    filepath = os.path.join(tmpdir, 'sample')
+    sample_df_pandas.to_csv(filepath, index=False)
+
+    no_type_error = "Content type could not be inferred. Please specify content_type and try again."
+    with pytest.raises(RuntimeError, match=no_type_error):
+        ww.read_file(filepath=filepath)
```
Comment (thehomebrewnerd) [inline woodwork/tests/test_utils.py:182]: Same as `test_read_file_errors_no_content_type` above?

**Review by thehomebrewnerd — COMMENTED** [review]
Looks like we have a couple lines that are uncovered still, and one test that appears to be duplicated.

File: woodwork/utils.py [hunk 2]
```diff
@@ -92,13 +100,24 @@ def read_csv(filepath=None,
...
+    if content_type is None:
+        inferred_type, encoding = guess_type(filepath)
+        if inferred_type is None:
+            raise RuntimeError('Content type could not be inferred. Please specify content_type and try again.')
+        if encoding is not None and 'encoding' not in kwargs:
+            kwargs['encoding'] = encoding
+        content_type = inferred_type
+
+    if content_type not in type_to_read_func_map:
+        raise RuntimeError('Reading from content type {} is not currently supported'.format(content_type))
```
[the remark this answers is not in the document]
Comment (gsheni) [reply woodwork/utils.py:118]: added

File: woodwork/utils.py [hunk 3]
```diff
@@ -92,13 +100,24 @@ def read_csv(filepath=None,
...
-        pd.DataFrame: DataFrame created from the specified CSV file with Woodwork typing information initialized.
+        pd.DataFrame: DataFrame created from the specified file with Woodwork typing information initialized.
     """
-    dataframe = pd.read_csv(filepath, **kwargs)
+    if content_type is None:
+        inferred_type, encoding = guess_type(filepath)
+        if inferred_type is None:
+            raise RuntimeError('Content type could not be inferred. Please specify content_type and try again.')
+        if encoding is not None and 'encoding' not in kwargs:
+            kwargs['encoding'] = encoding
```
[the remark this answers is not in the document]
Comment (gsheni) [reply woodwork/utils.py:114]: added

File: docs/source/release_notes.rst [hunk 4]
```diff
@@ -30,8 +31,10 @@ Future Release
 
 Breaking Changes
 ++++++++++++++++
-    * Woodwork tables can no longer be saved using to disk ``df.ww.to_csv``, ``df.ww.to_pickle``, or 
+    * Woodwork tables can no longer be saved using to disk ``df.ww.to_csv``, ``df.ww.to_pickle``, or
       ``df.ww.to_parquet``. Use ``df.ww.to_disk`` instead.
+    * The ``read_csv`` function has been renamed to ``read_file``.
```
[the remark this answers is not in the document]
Comment (gsheni) [reply docs/source/release_notes.rst]: fixed

File: woodwork/tests/test_utils.py (hunk 1 above)
Comment (gsheni) [reply woodwork/tests/test_utils.py:192]: renamed and changed to `test_read_file_errors_unsupported`

**Comment by gsheni** [comment]
Decided to not use the `encoding` returned from `guess_type` as I couldn't find a use-case where this would be currently needed.

File: woodwork/tests/test_utils.py [hunk 5]
```diff
@@ -173,6 +173,16 @@ def test_read_file_errors_no_content_type(sample_df_pandas, tmpdir):
...
+    filepath = os.path.join(tmpdir, 'sample')
+    sample_df_pandas.to_feather(filepath)
+
+    content_type = "application/feather"
+    not_supported_error = "Reading from content type {} is not currently supported".format(content_type)
+    with pytest.raises(RuntimeError, match=not_supported_error):
+        ww.read_file(filepath=filepath, content_type=content_type)
+
+
 def test_read_file_errors_unhandled_content_type(sample_df_pandas, tmpdir):
```
Comment (thehomebrewnerd) [inline woodwork/tests/test_utils.py:186]: Isn't this test `test_read_file_errors_unhandled_content_type` still identical to `test_read_file_errors_no_content_type` or am I missing something?

Comment (gsheni) [reply woodwork/tests/test_utils.py]: Yep, removed the duplicate test.

**Review by thehomebrewnerd — APPROVED** [review]
Looks good!

**Review by tamargrey — APPROVED** [review]
looks good to me!

**Pull request merged**